app/src/main/java/com/android/voyageur/ui/trip/photos/PhotosScreen.kt

```diff
@@ -0,0 +1,313 @@
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import coil.compose.rememberAsyncImagePainter
+import com.android.voyageur.R
+import com.android.voyageur.model.trip.TripsViewModel
+import com.android.voyageur.model.user.UserViewModel
+import com.android.voyageur.ui.gallery.PermissionButtonForGallery
+import com.android.voyageur.ui.navigation.BottomNavigationMenu
+import com.android.voyageur.ui.navigation.LIST_TOP_LEVEL_DESTINATION
+import com.android.voyageur.ui.navigation.NavigationActions
+import com.android.voyageur.utils.ConnectionState
+import com.android.voyageur.utils.connectivityState
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+
+/**
+ * Composable function to display the Photos screen of the app. This screen displays the photos
+ * associated with a selected trip and allows the user to add, view, and delete photos.
+ *
+ * @param tripsViewModel [TripsViewModel] The ViewModel responsible for managing trips and their
+ *   associated data.
+ * @param navigationActions [NavigationActions] Used for handling navigation between screens.
+ * @param userViewModel [UserViewModel] The ViewModel for managing user-related data.
+ */
+@SuppressLint("StateFlowValueCalledInComposition", "MutableCollectionMutableState")
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalCoroutinesApi::class)
+@Composable
+fun PhotosScreen(
+    tripsViewModel: TripsViewModel,
+    navigationActions: NavigationActions,
+    userViewModel: UserViewModel
+) {
+  val trip by tripsViewModel.selectedTrip.collectAsState()
+  val photos = trip?.photos ?: emptyList()
+  var selectedPhoto by remember { mutableStateOf<String?>(null) }
+  val context = LocalContext.current
+  val status by connectivityState()
+  val isConnected = status === ConnectionState.Available
+
+  Scaffold(
+      modifier = Modifier.testTag("photosScreen"),
+      floatingActionButton = {
+        // Button to allow the user to add photos from the gallery
+        PermissionButtonForGallery(
+            onUriSelected = { uri ->
+              uri?.let {
+                if (isConnected) {
+                  // Upload the image to Firebase
+                  val imageUriParsed = Uri.parse(uri.toString())
+                  tripsViewModel.uploadImageToFirebase(
+                      uri = imageUriParsed,
+                      onSuccess = { downloadUrl ->
+                        tripsViewModel.addPhotoToTrip(downloadUrl)
```

Comment:
Extract the texts for the toast to the `strings.xml` file please

Reply:
Done